src/infill/SierpinskiFill.h

```diff
@@ -0,0 +1,466 @@
+ * |  ↑ \                       |  ↑ \                                    .
+ * |  ↑  C\  subdivides into    |C_↑__A\                                  .
+ * |  ↑   /                     |C ↑  B/                                  .
+ * |  ↑ /                       |  ↑ /                                    .
+ * |A /                         |A /   AB_TO_BC                           .
+ * |/                           |/                                        .
+ *                                                                        .
+ * |\                           |\                                        .
+ * |B \  AC_TO_AB               |B \   AC_TO_BC                           .
+ * |    \                       |↖   \                                    .
+ * |↖    C\  subdivides into    |C_↖__A\                                  .
+ * |  ↖   /                     |C ↑  B/                                  .
+ * |    /                       |  ↑ /                                    .
+ * |A /                         |A /   AB_TO_BC                           .
+ * |/                           |/                                        .
+ *                                                                        .
+ * |\                           |\                                        .
+ * |B \  AB_TO_BC               |B \   AC_TO_AB                           .
+ * |  ↗ \                       |  ↑ \                                    .
+ * |↗    C\  subdivides into    |C_↑__A\                                  .
+ * |      /                     |C ↗  B/                                  .
+ * |    /                       |↗   /                                    .
+ * |A /                         |A /   AC_TO_BC                           .
+ * |/                           |/                                        .
+ * 
+ * 
+ * Each triangle is associated with the length of the Sierpinski dual curve going through the triangle.
+ * The realized density of a triangle is calculated using the length of the Sierpinski segment,
+ * the line width and the area of the triangle.
+ * The realized density is compared with the requested average density over the triangle according to
+ * the volumetric density specification of the user.
+ * 
+ * 
+ * 
+ * === BALANCING ===
+ * 
+ * If the total realized length of all children of a triangle is less than the average requested length of the parent triangle,
+ * the parent triangle would definitely like to be subdivided.
+ * (When dithering, we may subdivide at a lower threshold and a constraint may cause such a triangle not to be subdivided after all. See CONSTRAINTS.)
+ * However, it might occur that one child then ends up with a requested lengthlower than its realized length,
+ * namely when the other child has a surplus of requested length.
+ * The error thus induced should be recorded.
+ * Value will be transported from the child with a surplus to the child with a lack of requested length.
```

Comment:
Clarified in b23cca66e03858a996c1631f67d2f56e61902d7c

This is not about a total error over the whole pattern. I dont really see where your misunderstanding comes from.

added:
```
 * Example:
 * Parent cell realizes 10mm filament, but child 1 requests 100mm and child 2 requests 2mm.
 * Subdivision would lead to two cells with 14mm filament realized.
 * Cell 2 therefore obtains 12mm filament from cell 1, where 12mm is recorded as an error.
```